### PR TITLE
Gpr malloc is all

### PR DESCRIPTION
**Pull request by nicolasnoble**
cc @jboeuf

### CODE REVIEW

**Comment by nicolasnoble** [comment]
And I am missing a lot of imbalanced frees...

File: src/core/tsi/fake_transport_security.c [hunk 1]
```diff
@@ -134,12 +134,12 @@ static void tsi_fake_frame_reset(tsi_fake_frame *frame, int needs_draining) {
 static int tsi_fake_frame_ensure_size(tsi_fake_frame *frame) {
   if (frame->data == NULL) {
     frame->allocated_size = frame->size;
-    frame->data = malloc(frame->allocated_size);
+    frame->data = gpr_malloc(frame->allocated_size);
     if (frame->data == NULL) return 0;
```
Comment (jboeuf) [inline src/core/tsi/fake_transport_security.c]: we can probably remove that now

File: src/core/tsi/fake_transport_security.c [hunk 2]
```diff
@@ -134,12 +134,12 @@ static void tsi_fake_frame_reset(tsi_fake_frame *frame, int needs_draining) {
...
   } else if (frame->size > frame->allocated_size) {
     unsigned char *new_data = realloc(frame->data, frame->size);
```
Comment (jboeuf) [inline src/core/tsi/fake_transport_security.c]: gpr_realloc. Please make sure to fix all these. That could be dangerous.

File: src/core/tsi/fake_transport_security.c [hunk 3]
```diff
@@ -160,7 +160,7 @@ static tsi_result fill_frame_from_bytes(const unsigned char *incoming_bytes,
   if (frame->needs_draining) return TSI_INTERNAL_ERROR;
   if (frame->data == NULL) {
     frame->allocated_size = TSI_FAKE_FRAME_INITIAL_ALLOCATED_SIZE;
-    frame->data = malloc(frame->allocated_size);
+    frame->data = gpr_malloc(frame->allocated_size);
     if (frame->data == NULL) return TSI_OUT_OF_RESOURCES;
```
Comment (jboeuf) [inline src/core/tsi/fake_transport_security.c]: Remove NULL check.

**Comment by jboeuf** [comment]
Thanks for the cleanup. Please make sure to sanitize the realloc calls as well.

**Comment by nicolasnoble** [comment]
That one is ready to go.

File: src/core/lib/profiling/basic_timers.c [hunk 4]
```diff
@@ -208,7 +208,7 @@ static void init_output() {
 }
 
 static void rotate_log() {
-  gpr_timer_log *new = malloc(sizeof(*new));
+  gpr_timer_log *new = gpr_malloc(sizeof(*new));
```
Comment (ctiller) [inline src/core/lib/profiling/basic_timers.c]: This can be called by gpr_malloc.

Comment (nicolasnoble) [inline src/core/lib/profiling/basic_timers.c]: Done.

**Pull request merged**